### DISCUSSION TITLE
WASM target?

### DISCUSSION THREAD

**Discussion by Zireael07** [edubart/nelua-lang#11] [Q&A] [answered]
Quote from nelua's landing page:
> Any platform that a C99 compiler targets the language is capable of targeting so the language can take advantage of highly optimized compilers such as GCC and Clang, thus generating very efficient native code.

Clang can output WASM (via LLVM), does that mean WASM target is possible?

**Comment by edubart** [comment, marked as answer]
It is, actually this game made in Nelua compiles to WASM already (using Emscripten).

https://edubart.github.io/nelua-game2048/
https://github.com/edubart/nelua-game2048

**Reply by ArnaudValensi** [reply]
There is another example here: 

https://github.com/ArnaudValensi/nelua-sdl2-emscripten-docker

**Comment by Zireael07** [comment]
Great to know! This, however, means that Nelua could use a place to showcase projects using Nelua in practice (I was under the impression that it was yet another programming language experiment that has no real life uses, like 90% of indie programming languages)
Either on a github wiki or on the main page :)

**Comment by edubart** [comment]
@Zireael07 Thanks for the suggestion, at the moment this place exists in the `showcase` channel in the discord chat. I will make a page with many examples and uses and then link in the main page once I feel more comfortable with the language features and documentation. Right now important features is yet to be implemented (specially for users expecting Lua goodies) and more documentation is needed, so I don't want encourage anyone to use it yet, only the enthusiasts and they generally are on the discord chat.

**Comment by Zireael07** [comment]
You didn't link the discord anywhere that I can see, either ;) I'd love to join in!

**Comment by edubart** [comment]
The link is on the chat badge in the main page, it is https://discord.com/invite/7aaGeG7

**Comment by erlend-sh** [comment]
The fact that Nelua can target WASM seems deeply un-advertised. Given WASM’s rise, I think it’s one of the most appealing properties of Nelua.

Maybe add it here? https://github.com/appcypher/awesome-wasm-langs
